plugins/repository-s3/src/test/java/org/opensearch/repositories/s3/utils/HttpRangeUtilsTests.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.repositories.s3.utils;
+
+import software.amazon.awssdk.core.exception.SdkException;
+
+import org.opensearch.test.OpenSearchTestCase;
+
+public final class HttpRangeUtilsTests extends OpenSearchTestCase {
+
+    public void testFromHttpRangeHeader() {
+        String headerValue = "bytes 0-10/200";
+        Long offset = HttpRangeUtils.getStartOffsetFromRangeHeader(headerValue);
+        assertEquals(0L, offset.longValue());
+
+        headerValue = "bytes 0-10/*";
+        offset = HttpRangeUtils.getStartOffsetFromRangeHeader(headerValue);
+        assertEquals(0L, offset.longValue());
+
+        final String invalidHeaderValue = "bytes */*";
```

Comment:
How about `"bytes 9223372036854775808-9223372036854775809/0"`?

Reply:
I think that would break a lot of assumptions within our code and the S3 APIs. The API contract for object length is limited by `Long` - https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/services/s3/model/GetObjectAttributesResponse.html#objectSize()
Given that the content window has to be within the bounds of `0 < content length < 9223372036854775808`, we should never hit this scenario.